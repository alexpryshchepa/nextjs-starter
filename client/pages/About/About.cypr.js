describe('About page', () => {
  it('Checks if url matches', () => {
    cy.visit('/about');

    cy
      .url()
      .should('include', '/about');
  });
});
