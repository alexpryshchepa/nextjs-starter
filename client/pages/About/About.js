import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { withNamespaces, Link } from 'i18n';
import Loader from 'client/components/Loader';
import s from './About.scss';

export const About = ({ t, docs }) => {
  const renderDocs = () => {
    if (docs.isLoaded) {
      return <div dangerouslySetInnerHTML={{ __html: docs.data }} />;
    } if (docs.hasError) {
      return <span className={s.error}>{t('error')}</span>;
    }

    return (
      <div className={s.loader}>
        <Loader />
      </div>
    );
  };

  return (
    <div className={s.root}>
      <Link prefetch href="/">
        <a className={s.link}>{t('link')}</a>
      </Link>
      <h1 className={s.title}>{t('title')}</h1>
      <div className={s.docs}>
        {renderDocs()}
      </div>
    </div>
  );
};

About.propTypes = {
  t: PropTypes.func.isRequired,
  docs: PropTypes.shape({
    isLoaded: PropTypes.bool,
    hasError: PropTypes.bool,
    data: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.instanceOf(null),
    ]),
  }).isRequired,
};

export default compose(
  connect(state => ({
    docs: state.docs,
  })),
  withNamespaces('about'),
)(About);
