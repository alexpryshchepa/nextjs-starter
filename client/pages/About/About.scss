@import '~client/styles/variables';
@import '~client/styles/mixins';

.root {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  padding: 64px;

  @include tablet {
    padding: 32px;
  }

  @include mobile {
    padding: 24px;
  }
}

.link {
  text-align: center;
  margin-bottom: 24px;
  color: $color-success;
}

.title {
  margin-bottom: 24px;
  font-size: 24px;
  line-height: 36px;
  text-align: center;
}

.docs {
  max-width: 100%;

  :global {
    @import '~github-markdown-css';
  }
}

.error {
  color: $color-error;
}

.loader {
  padding: 24px 0;
}
