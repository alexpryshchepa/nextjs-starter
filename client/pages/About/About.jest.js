/* eslint-env jest */
import { shallow } from 'enzyme';
import React from 'react';
import { About } from './About';

describe('With Enzyme', () => {
  it('Checks title text and snapshot matching', () => {
    const component = shallow(
      <About
        t={key => key}
        docs={{
          loaded: false,
          error: false,
          docs: null,
        }}
      />,
    );

    expect(
      component.find('h1').text(),
    ).toEqual('title');

    expect(
      component.find('div.loader'),
    ).toHaveLength(1);

    expect(component).toMatchSnapshot();
  });
});
