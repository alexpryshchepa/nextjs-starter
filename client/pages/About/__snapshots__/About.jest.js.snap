// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page component Checks snapshot matching 1`] = `ShallowWrapper {}`;

exports[`about page component user interface checks snapshot matching 1`] = `ShallowWrapper {}`;
